test/HttpTest.cpp

```diff
@@ -64,8 +68,22 @@ TEST(HttpServer, HttpTest) {
         "Could not start test HTTP server on any of the ports");
   }();
 
-  // Run the server in its own thread.
-  std::jthread httpServerThread([&]() { httpServer.run(); });
+  auto& httpServer = *httpServerPtr;
+
+  // Run the server in its own thread. Wait for 100ms until the server is
+  // up (it should be up immediately).
```

Comment:
```suggestion
  // up (it should be up immediately). We have to move the server into the thread because it might have to outlive the current thread if we detach the thread in the unlikely case of not being able to start the server.
```